src/registrar/tests/test_models_domain.py

```diff
@@ -2597,59 +2619,151 @@ def test_analyst_deletes_domain(self):
+        self.domain_with_contacts.save()
+
+        # Check that the host and contacts are deleted
+        self.mockedSendFunction.assert_has_calls(
+            [
+                call(
+                    commands.UpdateDomain(
+                        name='freeman.gov',
+                        add=[common.Status(state=Domain.Status.CLIENT_HOLD, description='', lang='en')],
+                        rem=[],
+                        nsset=None,
+                        keyset=None,
+                        registrant=None,
+                        auth_info=None,
+                    ),
+                    cleaned=True,
+                ),
+                call(
+                    commands.InfoDomain(name='freeman.gov', auth_info=None),
+                    cleaned=True,
+                ),
+                call(
+                    commands.InfoHost(name='fake.host.com'),
+                    cleaned=True,
+                ),
+                call(
+                    commands.UpdateDomain(
+                        name='freeman.gov',
+                        add=[],
+                        rem=[common.HostObjSet(hosts=['fake.host.com'])],
+                        nsset=None,
+                        keyset=None,
+                        registrant=None,
+                        auth_info=None,
+                    ),
+                    cleaned=True,
+                ),
+                call(
+                    commands.DeleteHost(name='fake.host.com'),
+                    cleaned=True,
+                ),
+                call(
+                    commands.UpdateDomain(
+                        name='freeman.gov',
+                        add=[],
+                        rem=[common.DomainContact(contact='adminContact', type='admin')],
+                        nsset=None,
+                        keyset=None,
+                        registrant=None,
+                        auth_info=None,
+                    ),
+                    cleaned=True,
+                ),
+                call(
+                    commands.DeleteContact(id='adminContact'),
+                    cleaned=True,
+                ),
+                call(
+                    commands.UpdateDomain(
+                        name='freeman.gov',
+                        add=[],
+                        rem=[common.DomainContact(contact='techContact', type='tech')],
+                        nsset=None,
+                        keyset=None,
+                        registrant=None,
+                        auth_info=None,
+                    ),
+                    cleaned=True,
+                ),
+                call(
+                    commands.DeleteContact(id='techContact'),
+                    cleaned=True,
+                ),
+                call(
+                    commands.DeleteDomain(name='freeman.gov'),
+                    cleaned=True,
+                ),
+            ],
+            any_order=True,
```

Comment:
The ordering of these matters in some cases, but the database doesn't always return things in the same order so for things like the contacts it can break the test. I'll break that part out though, because in general you have the right idea and I had meant to do so anyway.